### PR TITLE
[Feature] Add version check in case of backend update

### PR DESCRIPTION
**Pull request by llanghammer-sap**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [ ] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description
This PR introduces a check to the backend to see if the required Backend version is higher than the installed version. This functionality is introduced to inform the user about a update in case there are breaking changes from the backend side. 

Currently the check is triggered during the setup of the UI in the SceneDelegate to display the resulting Alert Controller over the first ViewController. If the update is required for the app to keep working the Alert is not dismissable. 

## How to Test
You can manually specify versions in the AppUpdateCheckerHelper.swift file in the checkAppVersionDialog() function. If the minVersion is higher than the current version of the app a non dismissable Dialog should be triggered, if only the latestVersion is higher a one time dialog on app start is triggered.

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+				UIApplication.shared.open(url, options: [:], completionHandler: nil)
+		}))
+		switch type {
+		case .update:
+			alert.addAction(UIAlertAction(title: NSLocalizedString("Später aktualisieren", comment: "Remind me later"), style: .default, handler: { _ in
+				self.setObserver(vc: vc)
+				//Do nothing
+			}))
+		case .forceUpdate:
+			alert.message = "Um die Applikation weiter zu nutzen müssen sie eine neue Version installieren"
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift:89]: Loc?

File: src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	private func removeObserver() {
+		NotificationCenter.default.removeObserver(applicationDidBecomeActiveObserver as Any)
+        applicationDidBecomeActiveObserver = nil
+	}
+
+	func createAlert(_ type: UpdateAlertType, vc: UIViewController?) -> UIAlertController? {
+		let alert = UIAlertController(title: "Akutalisierung verfügbar", message: "Es gibt eine neue Aktualisierung für die Applikation", preferredStyle: .alert)
+		alert.addAction(UIAlertAction(title: NSLocalizedString("Aktualisieren", comment: "Default action"), style: .default, handler: { _ in
+			//TODO: Add correct App Store ID
+			guard let url: URL = URL(string: "itms-apps://itunes.apple.com/app/apple-store/") else {
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift:77]: I think we should store the fact that the user has already been asked in order to not ask every timr.

File: src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift (hunk 1 above)
Comment (llanghammer-sap) [reply src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift]: Added with https://github.com/corona-warn-app/cwa-app-ios/pull/269/commits/02fc0a9450c15753f564d41a293eedafacb47084

Only german texts at the moment.

File: src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift (hunk 2 above)
Comment (llanghammer-sap) [reply src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift:80]: I just checked with Chris, the first implementation should display the non-force Update on every app start. 

I fixed a Bug a small bug where the non-force Update kept appearing after ech switch to the app, which was not the exspected behaviour.

**Review by inf2381 — APPROVED** [review]

**Review by ChristianKienle — APPROVED** [review]

**Pull request merged**